### PR TITLE
[SYCL] Remove customizations from legacy pipeine

### PR DESCRIPTION
**Pull request by AlexeySachkov**
Moved definition of `SYCLOptimizationMode` option into new pass manager. This change is inspired by https://reviews.llvm.org/D137915 in which community refactors the code in preparation for complete removal of a legacy pass manager.

Legacy pass manager is not available to user since 2978d026819b82f77eef42d92b3b53919f375bd8 and 69b2b7282e92a1b576b7bd26f3b16716a5027e8e

### CODE REVIEW

**Comment by AlexeySachkov** [comment]
My understanding is that we are not using legacy pipeline anymore and therefore this should be a non-functional change

**Comment by bader** [comment]
Please, fix pre-commit failures.

**Comment by AlexeySachkov** [comment]
> I'm okay with removing legacy, but, please, make sure there is no fallback option which enables legacy pass manager flow.

I've added that info into the description: flags to enable legacy pass manager flow were made no-op in release 15 and later they were removed completely

**Review by KornevNikita — APPROVED** [review]

**Comment by AlexeySachkov** [comment]
@intel/dpcpp-tools-reviewers, friendly ping.

Failed `sort.cpp` test is not related and it was already disabled for CUDA & HIP

File: llvm/lib/Transforms/IPO/PassManagerBuilder.cpp [hunk 1]
```diff
@@ -129,15 +124,7 @@ void PassManagerBuilder::addFunctionSimplificationPasses(
   MPM.add(
       createCFGSimplificationPass(SimplifyCFGOptions().convertSwitchRangeToICmp(
           true)));                            // Merge & remove BBs
-  // FIXME: re-association increases variables liveness and therefore register
-  // pressure.
-  if (!SYCLOptimizationMode)
```
[the remark this answers is not in the document]
Comment (AlexeySachkov) [reply llvm/lib/Transforms/IPO/PassManagerBuilder.cpp:134]: Right

**Review by sarnex — APPROVED** [review]

**Comment by AlexeySachkov** [comment]
HIP failures are known and tracked in #7634

**Pull request merged**